### PR TITLE
BUG: Fix output for two-tensor tractography fiber selection

### PR DESCRIPTION
**Pull request by zhangfanmark**
Mantis ID: 0004042

For two-tensor tractography, FiberBundleLabelSelect outputs only one
tensor. This change outputs all tensor information from the selected
fibers of the input file.

### CODE REVIEW

File: Modules/CLI/FiberBundleLabelSelect/FiberBundleLabelSelect.cxx [hunk 1]
```diff
@@ -292,26 +291,26 @@ int main( int argc, char * argv[] )
   outFibersCellArray->Allocate(numNewPts+numNewCells);
   outFibers->SetLines(outFibersCellArray.GetPointer());
 
-  //outFibersCellArray->SetNumberOfCells(numNewCells);
-  //outFibersCellArray = outFibers->GetLines();
+  // If the input has point data, including tensors or scalar arrays, copy them to the output.
+  // Currently this ignores cell data, which may be added in the future if needed.
+  // Check for point data arrays to keep and allocate them.
+  int numberArrays = input->GetPointData()->GetNumberOfArrays();
+  //std::cerr << "Input data has " << numberArrays << " point data arrays." << std::endl;
```
Comment (pieper) [inline Modules/CLI/FiberBundleLabelSelect/FiberBundleLabelSelect.cxx]: Better to remove any commented out debug statements before finalizing the code.

**Comment by pieper** [comment]
Looks like a nice fix.  It would be good to have a test so that people will see how this fix is working and prevent any regression in the future.

File: Modules/CLI/FiberBundleLabelSelect/FiberBundleLabelSelect.cxx [hunk 2]
```diff
@@ -292,26 +291,26 @@ int main( int argc, char * argv[] )
...
-    newTensors->SetNumberOfComponents(9);
-    newTensors->Allocate(9*numNewPts);
-    outFibers->GetPointData()->SetTensors(newTensors);
-    newTensors = static_cast<vtkFloatArray *> (outFibers->GetPointData()->GetTensors());
+    //std::cerr << "Found array " << arrayIdx << std::endl;
+    vtkDataArray *oldArray = input->GetPointData()->GetArray(arrayIdx);
+    vtkSmartPointer<vtkFloatArray> newArray = vtkSmartPointer<vtkFloatArray>::New();
+    newArray->SetNumberOfComponents(oldArray->GetNumberOfComponents());
+    newArray->SetName(oldArray->GetName());
+    newArray->Allocate(newArray->GetNumberOfComponents()*numNewPts);
```
Comment (jcfr) [inline Modules/CLI/FiberBundleLabelSelect/FiberBundleLabelSelect.cxx]: Would it make sense to DeepCopy the array doing something like this:

```
vtkNew<vtkFloatArray> newArray;
newArray->DeepCopy(oldArray);
outFibers->GetPointData()->AddArray(newArray.GetPointer());
```

File: Modules/CLI/FiberBundleLabelSelect/FiberBundleLabelSelect.cxx (hunk 1 above)
Comment (zhangfanmark) [inline Modules/CLI/FiberBundleLabelSelect/FiberBundleLabelSelect.cxx]: Thanks Steve! I'll remove those comments.

File: Modules/CLI/FiberBundleLabelSelect/FiberBundleLabelSelect.cxx (hunk 2 above)
Comment (zhangfanmark) [inline Modules/CLI/FiberBundleLabelSelect/FiberBundleLabelSelect.cxx]: Thanks, JC! This should be better and I'll try it later.

**Comment by jcfr** [comment]
@zhangfanmark Out of curiosity, did you have a chance to look into adding a test ?

**Comment by zhangfanmark** [comment]
Hi JC @jcfr, 
CC: Steve @pieper

Yes, I did plan to add the test along with the DeepCopy you mentioned. But, I was on travel during the last week and have not got time for the updating. I'll be back in two weeks and will update the codes once I am back to Boston.

Regards,
Fan

**Comment by jcfr** [comment]
@zhangfanmark Is there any update on this ? 

Considering we will be releasing very soon, would be great to get this fixed. 

Let me know if you have any any questions, 

If you would like to see an example of test. See [r24673](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=24673) added by @orbe77 

Cc: @pieper @ljod

**Comment by ljod** [comment]
Thanks JC! Orjan and Fan will talk tomorrow morning about setting up this test.

Sent by Outlook<http://taps.io/outlookmobile> for Android



On Wed, Oct 28, 2015 at 9:25 PM -0700, "Jean-Christophe Fillion-Robin" <notifications@github.com<mailto:notifications@github.com>> wrote:


@zhangfanmark<https://github.com/zhangfanmark> Is there any update on this ?

Considering we will be releasing very soon, would be great to get this fixed.

Let me know if you have any any questions,

If you would like to see an example of test. See r24673<http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=24673> added by @orbe77<https://github.com/orbe77>

Cc: @pieper<https://github.com/pieper> @ljod<https://github.com/ljod>

—
Reply to this email directly or view it on GitHub<https://github.com/Slicer/Slicer/pull/349#issuecomment-152072765>.


The information in this e-mail is intended only for the person to whom it is
addressed. If you believe this e-mail was sent to you in error and the e-mail
contains patient information, please contact the Partners Compliance HelpLine at
http://www.partners.org/complianceline . If the e-mail was sent to you in error
but does not contain patient information, please contact the sender and properly
dispose of the e-mail.

**Comment by jcfr** [comment]
Closing this PR. Superseded by #395

**Pull request closed without merging**